src/Microsoft.VisualStudio.Services.Agent/StreamString.cs

```diff
@@ -50,8 +50,12 @@ public StreamString(Stream ioStream)
         public async Task<string> ReadStringAsync(CancellationToken cancellationToken)
         {            
             Int32 len = await ReadInt32Async(cancellationToken);
-            cancellationToken.ThrowIfCancellationRequested();            
-            if (len <= 0 || len > MaxStringSize)
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Same comment to line 77:
cancellationToken.ThrowIfCancellationRequested();
return streamEncoding.GetString(inBuffer);